### PR TITLE
[WIP] Remove all enc_utf8, enc_dbcs, and has_mbyte conditions in screen.c

### PR DESCRIPTION
**Pull request by alekulyn**
This is a migration of the enc_utf8 and enc_dbcs changes that @romgrk made in #5177 into one PR.  Each class of changes is separated into its own commit.
@bfredl

### CODE REVIEW

**Comment by alekulyn** [comment]
Temporary close while I fix some errors.

**Comment by bfredl** [comment]
There is no need to close it because there might be errors, it is implied by `[WIP]`

**Comment by bfredl** [comment]
Also it might make sense to assume `has_mbyte=true`, and maybe also `enc_dbcs=0`, in the same PR, at least in places close to where you already remove `enc_utf8`, to avoid going over the same code twice.

File: src/nvim/screen.c [hunk 1]
```diff
@@ -5293,11 +5224,10 @@ void screen_puts_len(char_u *text, int textlen, int row, int col, int attr)
   int need_redraw;
 
   const bool l_has_mbyte = has_mbyte;
-  const bool l_enc_utf8 = enc_utf8;
   const int l_enc_dbcs = enc_dbcs;
 
-  assert((l_has_mbyte == (l_enc_utf8 || l_enc_dbcs))
-         && !(l_enc_utf8 && l_enc_dbcs));
+  assert((l_has_mbyte == l_enc_dbcs)
```
Comment (bfredl) [inline src/nvim/screen.c]: This assert can be completely removed in any case.

**Comment by bfredl** [comment]
There is also a bit of linter errors to fix: https://travis-ci.org/neovim/neovim/jobs/181031704 (ignore the one in `buffer.c`, it should dissappear by itself your next push).
you can run `make clint LINT_FILE=src/nvim/screen.c` locally for quicker feedback.

Also please remove `has_mbyte` when close to the existing changes, so we won't need to review two very similar restructurings to the same blocks of code.

**Comment by justinmk** [comment]
Closing in favor of #6609.

**Pull request closed without merging**